homeassistant/components/blebox/__init__.py

```diff
@@ -0,0 +1,112 @@
+"""The BleBox devices integration."""
+import asyncio
+import logging
+
+from blebox_uniapi.error import Error
+from blebox_uniapi.products import Products
+from blebox_uniapi.session import ApiHost
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_SETUP_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["cover"]
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the BleBox devices component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up BleBox devices from a config entry."""
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, platform)
+                for platform in PLATFORMS
+            ]
+        )
+    )
+
+    return unload_ok
+
+
+async def async_add_blebox(klass, method, hass, config, async_add, exception):
+    """Add a BleBox device from the given config."""
+    host = config[CONF_HOST]
+    port = config[CONF_PORT]
+    timeout = DEFAULT_SETUP_TIMEOUT
+
+    websession = async_get_clientsession(hass)
+    api_host = ApiHost(host, port, timeout, websession, hass.loop, _LOGGER)
+    try:
+        product = await Products.async_from_host(api_host)
+    except Error as ex:
+        _LOGGER.error("Identify failed at %s:%d (%s)", host, port, ex)
+        raise exception from ex
```

Comment:
```suggestion
        raise
```

Reply:
If its a connection error you can raise `ConfigEntryNotReady` to have this retried later

Reply:
I tried `ConfigEntryNotReady` at first, except it doesn't seem to work.

(That's why I added the parameter - so I could pick exceptions once I could figure out which scenario works in practice with which exception).

Instead, using PlatformNotReady works as expected (handled + logging just the error message + HA retries the setup multiple times).

Scenario:
1. I configure a device (user flow).
2. I disconnect the physical device from the network.
3. I restart HA
4. Exception backtrace is logged when ConfigEntryNotReady is used (raises ConfigEntryNoReady, which is never caught and no retries), but works as expected when PlatformNotReady is used (expected = only logs error, with no exception backtrace, then retries).

During HA startup, it seems the setup is handled in `entity_platform.py` which does not handle `ConfigEntryNotReady` exceptions. (I can't even figure out how to trigger a case where ConfigEntryNotReady is likewise handled with retries/etc.)

Reply:
https://developers.home-assistant.io/docs/creating_component_generic_discovery
"Most integrations consist of a single platform. And in that case, it's fine to just define that one platform. However, if you are going to add a second platform, you will want to centralize your connection logic. This is done inside the component (__init__.py)."

https://developers.home-assistant.io/docs/config_entries_index

You'll need to raise ConfigEntryNotReady in `async_setup_entry` in `__init__.py`